plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java

```diff
@@ -110,6 +110,9 @@ public static Type createTypeFromCoercer(TypeManager typeManager, HiveType fromH
         if (fromType instanceof VarcharType varcharType && toHiveType.equals(HIVE_DOUBLE)) {
             return Optional.of(new VarcharToDoubleCoercer(varcharType, coercionContext.treatNaNAsNull()));
         }
+        if (fromType instanceof VarcharType varcharType && toHiveType.equals(HIVE_FLOAT)) {
```

Comment:
potential follow-up: There's a lot of `if` statements which deal with

```
fromType instanceof VarcharType varcharType
```

Maybe group them inside a big if would make the code easier the read. This is just an opinion, it doesn't need necessarily to be implemented.

Reply:
I'm planning to perform a minor clean-up post the coercions